packages/expo-notifications/android/src/main/java/expo/modules/notifications/serverregistration/LastRegistrationInfo.kt

```diff
@@ -0,0 +1,25 @@
+package expo.modules.notifications.serverregistration
+
+import android.content.Context
+import android.content.SharedPreferences
+
+class LastRegistrationInfo(context: Context) {
+  companion object {
+    private const val PREFERENCES_NAME = "expo.modules.notifications.LastRegistrationInfo"
+
+    private const val LAST_REGISTRATION_INFO_KEY = "lastRegistrationInfo"
+  }
+
+  private val sharedPreferences: SharedPreferences? = context.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE)
```

Comment:
```suggestion
  private val sharedPreferences: SharedPreferences = context.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE)
```
we can safely assume that the `getSharedPreferences` return a non-null value here

Reply:
`getSharedPreferences` returns `SharedPreferences` instance to work with entries not an entry.

Reply:
I believe that context.getSharedPreferences can't return a null value - https://developer.android.com/reference/android/content/Context#getSharedPreferences(java.lang.String,%20int).

Moreover, if somehow the android returns a null as a SharedPreferences, it will be our least worry.